### PR TITLE
move portforward to client-go

### PR DESCRIPTION
**Pull request by deads2k**
Second to last one.  I had to split some tests like we did for apimachinery, but they still run in kubernetes.

@sttts

### CODE REVIEW

File: staging/src/k8s.io/client-go/tools/portforward/portforward.go [hunk 1]
```diff
@@ -27,12 +27,15 @@ import (
...
 	"sync"
 
+	"k8s.io/apimachinery/pkg/util/httpstream"
 	"k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/client-go/pkg/api"
-	"k8s.io/client-go/pkg/kubelet/server/portforward"
-	"k8s.io/client-go/pkg/util/httpstream"
 )
 
+// TODO move to API machinery and re-unify with kubelet/server/portfoward
```
Comment (sttts) [inline staging/src/k8s.io/client-go/tools/portforward/portforward.go]: This does not belong to apimachinery. It's very kube specific. But I am fine with duplicating it, with a two-way reference.

Comment (sttts) [inline staging/src/k8s.io/client-go/tools/portforward/portforward.go]: Or add a unit test in kube referencing both.

**Comment by sttts** [comment]
One comment about a TODO. Otherwise, lgtm. Will conflict with the soon-to-be-merged client-go sync.

**Comment by deads2k** [comment]
> Will conflict with the soon-to-be-merged client-go sync.

naturally.  I'd be disappointed if I didn't get to rebase my pulls at least three times for bazel these days.

File: staging/src/k8s.io/client-go/tools/portforward/portforward.go (hunk 1 above)
Comment (deads2k) [inline staging/src/k8s.io/client-go/tools/portforward/portforward.go]: > This does not belong to apimachinery. It's very kube specific. But I am fine with duplicating it, with a two-way reference.

If anything else wanted to do a port forward, wouldn't we encourage them to remain consistent?

**Comment by deads2k** [comment]
weird, test

**Pull request merged**